app/Models/Category.php

```diff
@@ -13,4 +13,26 @@ public function products()
     {
         return $this->hasMany(Product::class);
     }
+
+    /**
+     * A Category has Many Child Categories
+     */
+    public function children()
+    {
+        return $this->hasMany(Category::class, 'parent_id', 'id');
+    }
+
+    /**
+     * A Category has or not A Parent Category
+     */
+    public function parent()
+    {
+        return $this->belongsTo(Category::class, 'parent_id', 'id');
+    }
+
+
+    public function isParent()
+    {
+        return !$this->parent_id ? true : false; // if parent_id is null => is a Parent Category
```

Comment:
```suggestion
        return is_null($this->parent_id);
```
ntn cho ngắn gọn đc ko nhỉ?